deepchecks/vision/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -0,0 +1,173 @@
+from numbers import Number
+from typing import Dict, List
+
+from ignite.metrics import Metric
+
+from deepchecks.core import CheckResult, ConditionResult, DatasetKind
+from deepchecks.core.checks import ReduceMixin
+from deepchecks.core.condition import ConditionCategory
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.metrics import get_scorers_dict, metric_results_to_df
+
+__all__ = ['SingleDatasetPerformance']
+
+
+class SingleDatasetPerformance(SingleDatasetCheck, ReduceMixin):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+    alternative_scorers : Dict[str, Callable], default: None
+        An optional dictionary of scorer name to scorer functions.
+        If none given, using default scorers
+
+    """
+
+    def __init__(self,
+                 alternative_scorers : Dict[str, Metric] = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.user_scorers = alternative_scorers
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind.TRAIN):
+        """Initialize the metric for the check, and validate task type is relevant."""
+        self.scorers = get_scorers_dict(context.get_data_by_kind(dataset_kind), self.user_scorers)
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        for scorer in self.scorers.values():
+            scorer.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        results_dict = {}
+        for name, scorer in self.scorers.items():
+            result = scorer.compute()
+            results_dict[name] = [result] if isinstance(result, Number) else result
+        result_df = metric_results_to_df(results_dict, context.get_data_by_kind(dataset_kind))
+        return CheckResult(result_df)
+
+    def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
+        """Return the values of the metrics for the test dataset in {metric: value} format."""
+        # Find metrics that were reduced over the classes and replace the Class Name with None
+        is_reduced_metrics = check_result.value.groupby('Metric')['Class Name'].nunique() == 1
+        reduced_metrics = is_reduced_metrics[is_reduced_metrics].index.to_list()
+        check_result.value.loc[check_result.value.Metric.apply(lambda x: x in reduced_metrics), 'Class Name'] = None
+
+        # Dict keys are format metric_class
+        metric_class = check_result.value.loc[:, ['Metric', 'Class Name']].aggregate(lambda x:
```

Comment:
How does metrics that are originally aggregated will look like? for example f1_micro

Reply:
@Nadav-Barak  They will have just 1 line instead of line per class